sdk/include/opentelemetry/sdk/metrics/ungrouped_processor.h

```diff
@@ -31,7 +72,7 @@ class UngroupedMetricsProcessor : public MetricsProcessor
 
 private:
   bool stateful_;
-  std::unordered_map<std::string, sdkmetrics::AggregatorVariant> batch_map_;
+  std::unordered_map<KeyStruct, sdkmetrics::AggregatorVariant, KeyStruct_Hash> batch_map_;
```

Comment:
Do we need to ensure the `KeyStruct` is immutable (e.g. adding const decorator)?